management/prompt.go

```diff
@@ -49,3 +120,54 @@ func (m *PromptManager) SetCustomText(ctx context.Context, p string, l string, b
 	err = m.management.Request(ctx, "PUT", m.management.URI("prompts", p, "custom-text", l), &b, opts...)
 	return
 }
+
+// CreateCustomPartials creates new custom prompt partials for a given segment.
+//
+// See: https://auth0.com/docs/sign-up-prompt-customizations#use-the-api-to-edit-custom-prompts
+func (m *PromptManager) CreateCustomPartials(ctx context.Context, c *CustomPrompt, opts ...RequestOption) error {
+	if err := validateCustomPromptSegment(c.Segment); err != nil {
+		return err
+	}
+	return m.management.Request(ctx, "PUT", m.management.URI("prompts", string(c.Segment), "partials"), c, opts...)
+}
+
+// UpdateCustomPartials updates custom prompt partials for a given segment.
+//
+// See: https://auth0.com/docs/sign-up-prompt-customizations#use-the-api-to-edit-custom-prompts
+func (m *PromptManager) UpdateCustomPartials(ctx context.Context, c *CustomPrompt, opts ...RequestOption) error {
+	if err := validateCustomPromptSegment(c.Segment); err != nil {
+		return err
+	}
+	return m.management.Request(ctx, "PUT", m.management.URI("prompts", string(c.Segment), "partials"), c, opts...)
+}
+
+// ReadCustomPartials reads custom prompt partials for a given segment.
+//
+// See: https://auth0.com/docs/sign-up-prompt-customizations#use-the-api-to-edit-custom-prompts
+func (m *PromptManager) ReadCustomPartials(ctx context.Context, segment CustomPromptSegment, opts ...RequestOption) (c *CustomPrompt, err error) {
+	if err := validateCustomPromptSegment(segment); err != nil {
+		return nil, err
+	}
+	c = &CustomPrompt{Segment: segment}
+	err = m.management.Request(ctx, "GET", m.management.URI("prompts", string(segment), "partials"), c, opts...)
+	return
+}
+
+// DeleteCustomPartials deletes custom prompt partials for a given segment.
+//
+// See: https://auth0.com/docs/sign-up-prompt-customizations#use-the-api-to-edit-custom-prompts
+func (m *PromptManager) DeleteCustomPartials(ctx context.Context, c *CustomPrompt, opts ...RequestOption) error {
```

Comment:
Nice one on this, I like the idea of keeping delete even though it the API doesn't strictly offer it.